src/infra/app-deployment.yml

```diff
@@ -0,0 +1,41 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hospital-appointment
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: hospital-appointment
+  template:
+    metadata:
+      labels:
+        app: hospital-appointment
+    spec:
+      containers:
+        - name: hospital-appointment
+          image: kermakovich/hospital-appointment
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 8080
+          env:
+            - name: DATASOURCE_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: pg-secrets
+                  key: DATASOURCE_USERNAME
+            - name: DATASOURCE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: pg-secrets
+                  key: DATASOURCE_PASSWORD
+            - name: DATASOURCE_URL
+              valueFrom:
+                secretKeyRef:
+                  name: pg-secrets
+                  key: DATASOURCE_URL
```

Comment:
Things like database url I'm not recommend to encrypt. Traceability of the project will be better, if you will use the configmap and env only the host name.